File: packages/sdk/dart/packages/core/lib/src/context_manager.dart
// Context manager for legacy isolateBy compatibility state.
//
// Dart equivalent of JS SDK's ContextManager. HTTP DB routing uses
// explicit `db(namespace, instanceId)` paths instead of a context header.

class ContextManager {
  Map<String, dynamic> _context = {};

  /// Set context keys. 'auth.id' is silently ignored (server extracts from JWT).
  void setContext(Map<String, dynamic> ctx) {
    final filtered = Map<String, dynamic>.from(ctx);
    filtered.remove('auth.id');
    _context = filtered;
  }

  /// Get current context.
  Map<String, dynamic> getContext() => Map.unmodifiable(_context);
}

File: packages/sdk/java/packages/core/src/main/java/dev/edgebase/sdk/core/ContextManager.java
// EdgeBase Java SDK — legacy isolateBy context manager.
//: auth.id is silently filtered — server extracts from JWT only.
package dev.edgebase.sdk.core;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

/**
 * Context manager for legacy isolateBy compatibility state.
 * HTTP DB routing uses explicit namespace and instance ID path segments.
 *
 * <p>
 * The {@code auth.id} key is silently filtered from context because the server
 * derives it from JWT claims only.
 */
public class ContextManager {
    private final Object lock = new Object();
    private Map<String, Object> context = Collections.emptyMap();

    public void setContext(Map<String, Object> ctx) {
        synchronized (lock) {
            Map<String, Object> filtered = new HashMap<>(ctx);
            filtered.remove("auth.id");
            context = Collections.unmodifiableMap(filtered);
        }
    }

    public Map<String, Object> getContext() {
        synchronized (lock) {
            return context;
        }
    }

    public void clearContext() {
        synchronized (lock) {
            context = Collections.emptyMap();
        }
    }
}
